src/extractor/scripting_environment.cpp

```diff
@@ -51,122 +52,137 @@ int luaErrorCallback(lua_State *state)
+             .def("query", &SourceContainer::getRasterDataFromSource)
+             .def("interpolate", &SourceContainer::getRasterInterpolateFromSource),
+         luabind::class_<const float>("constants")
+             .enum_("enums")[luabind::value("precision", COORDINATE_PRECISION)],
+
+         luabind::class_<ProfileProperties>("ProfileProperties")
+             .def(luabind::constructor<>())
+             .property("traffic_signal_penalty", &ProfileProperties::GetTrafficSignalPenalty,
+                       &ProfileProperties::SetTrafficSignalPenalty)
+             .property("u_turn_penalty", &ProfileProperties::GetUturnPenalty,
+                       &ProfileProperties::SetUturnPenalty)
+             .def_readwrite("allow_u_turn_at_via", &ProfileProperties::allow_u_turn_at_via),
+
+         luabind::class_<std::vector<std::string>>("vector")
+             .def("Add", static_cast<void (std::vector<std::string>::*)(const std::string &)>(
+                             &std::vector<std::string>::push_back)),
+
+         luabind::class_<osmium::Location>("Location")
+             .def<location_member_ptr_type>("lat", &osmium::Location::lat)
+             .def<location_member_ptr_type>("lon", &osmium::Location::lon),
+
+         luabind::class_<osmium::Node>("Node")
+             // .def<node_member_ptr_type>("tags", &osmium::Node::tags)
+             .def("location", &osmium::Node::location)
+             .def("get_value_by_key", &osmium::Node::get_value_by_key)
+             .def("get_value_by_key", &get_value_by_key<osmium::Node>)
+             .def("id", &osmium::Node::id),
+
+         luabind::class_<ExtractionNode>("ResultNode")
+             .def_readwrite("traffic_lights", &ExtractionNode::traffic_lights)
+             .def_readwrite("barrier", &ExtractionNode::barrier),
+
+         luabind::class_<ExtractionWay>("ResultWay")
+             // .def(luabind::constructor<>())
+             .def_readwrite("forward_speed", &ExtractionWay::forward_speed)
+             .def_readwrite("backward_speed", &ExtractionWay::backward_speed)
+             .def_readwrite("name", &ExtractionWay::name)
+             .def_readwrite("roundabout", &ExtractionWay::roundabout)
+             .def_readwrite("is_access_restricted", &ExtractionWay::is_access_restricted)
+             .def_readwrite("is_startpoint", &ExtractionWay::is_startpoint)
+             .def_readwrite("duration", &ExtractionWay::duration)
+             .property("forward_mode", &ExtractionWay::get_forward_mode,
+                       &ExtractionWay::set_forward_mode)
+             .property("backward_mode", &ExtractionWay::get_backward_mode,
+                       &ExtractionWay::set_backward_mode),
+         luabind::class_<osmium::Way>("Way")
+             .def("get_value_by_key", &osmium::Way::get_value_by_key)
+             .def("get_value_by_key", &get_value_by_key<osmium::Way>)
+             .def("id", &osmium::Way::id),
```

Comment:
Every now and then users ask for means to iterate over nodes in ways. Maybe we should register a [Way's .nodes()](http://docs.osmcode.org/libosmium/latest/classosmium_1_1Way.html) member function (and then probably also the `WayNodeList` for iteration) exposing this to the profiles.